File: viewjava/settings/src/main/java/pers/shawxingkwok/kdatastore/viewjava/settings/Theme.java
package pers.shawxingkwok.kdatastore.viewjava.settings;

import androidx.appcompat.app.AppCompatDelegate;

public enum Theme {
    FOLLOW_SYSTEM, DARK, LIGHT;

    public int getMode(){
        switch (this){
            case FOLLOW_SYSTEM: return AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM;
            case DARK: return AppCompatDelegate.MODE_NIGHT_YES;
            case LIGHT: return AppCompatDelegate.MODE_NIGHT_NO;
            default: throw new IllegalStateException("");
        }
    }
}

File: viewjava/settings/src/main/java/pers/shawxingkwok/kdatastore/viewjava/settings/Theme.kt
package pers.shawxingkwok.kdatastore.viewjava.settings

import androidx.appcompat.app.AppCompatDelegate

enum class Theme(val value: Int) {
    FOLLOW_SYSTEM(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM),
    DARK(AppCompatDelegate.MODE_NIGHT_YES),
    LIGHT(AppCompatDelegate.MODE_NIGHT_NO),
}